docs: add connector reference (en-US) (#89)
logto-io/docs: docs/docs/references/connectors/README.mdx

import ConnectorPlatformIntro from './fragments/_connector_platform_intro.mdx';
import ConnectorStorageIntro from './fragments/_connector_storage_intro.mdx';
import ConnectorTypeIntro from './fragments/_connector_type_intro.mdx';

# 🔌 Connectors

## Introduction

### What is a connector?

> _Connectors_ play a critical role in Logto. With their help, Logto enables end-users to use SMS-based or email-based passwordless registration and the capabilities of signing in with social accounts. With the increasing popularity of websites and applications, passwordless and social sign-ins allow users to avoid managing numerous accounts and passwords.

Reviewer: `enables end-users to use SMS-based or email-based ...` sounds not appropriate.

Author: deleted to make it more clear.

Sounds fascinating :hushed:, Uh? See [Configure connectors](../../recipes/configure-connectors/) if you want to set up an existing connector. If you can not find the connector you want to set up, you may develop those connectors by following the guides in [Create your connector](../../recipes/create-your-connector/).

## Compositions

_Connectors_ are stored both locally and remotely:

<ConnectorStorageIntro />

## Connector's local storage: _ConnectorMetadata_

### id

> _**id**_ is an _unique_ string-typed key to identify connector in Logto.

Reviewer: ```suggestion
_**id**_ is an _unique_ string-typed key to identify a connector in Logto.
```

It's assigned by the developers of each connector and will be uploaded to DB.

### target

_**target**_ is a lowercase string to distinguish the social identities source of the social connector.
For email and SMS connectors, _**target**s_ work more like a "human readable id".

For example, both _WeChat web connector_ and _WeChat native connector_'s _**target**_ are `wechat`. _Aliyun direct mail connector_'s _**target**_ is `aliyun-dm` and _Aliyun SMS connector_'s _**target**_ is `aliyun-sms`.

### type

_**type**_ should be one of the following connector types:

<ConnectorTypeIntro />

### platform

_**platform**_ should be one of the following values:

<ConnectorPlatformIntro />

### name

> _**name**_ is a Javascript/Typescript object whose keys are i18n country codes and values are connectors' display name.

Reviewer: maybe just object?

Author: Updated.

### logo

> _**logo**_ is an URL or relative path of connector's logo (light mode by default).

Reviewer: light mode by default stands for?

Author: deleted this to make it more clear and add descriptions in Notes.

### logoDark

> _**logo**_ is an _nullable_ URL or relative path of connector's dark mode logo.

Reviewer: _**logoDark**_ ?

:::note

> Usually, _**logo**s_ always exist, and _**logoDark**s_ only exist for some exceptional cases when the light mode logos may become obscure in dark mode.

Reviewer: I think just mentioning logo is required, logoDark is optional should be good enough. We display logo in both light and dark mode if no logoDark is provided.  

`some exceptional cases when the light mode logos may become obscure in dark mode` s` is objective. All we need to show is we support showing different logos under the different color themes.  Leave all the judgment to the users themselves.

Author: Good call, updated.

import ConnectorPlatformIntro from './fragments/_connector_platform_intro.mdx';
import ConnectorStorageIntro from './fragments/_connector_storage_intro.mdx';
import ConnectorTypeIntro from './fragments/_connector_type_intro.mdx';

# 🔌 Connectors

## Introduction

### What is a connector?

_Connectors_ play a critical role in Logto. With their help, Logto enables end-users to use passwordless registration or sign-in and the capabilities of signing in with social accounts. With the increasing popularity of websites and applications, passwordless and social sign-ins allow users to avoid managing numerous accounts and passwords.

:hushed: Sounds fascinating, Uh? See [Configure connectors](../../recipes/configure-connectors/) if you want to set up an existing connector. If you can not find the connector you want to set up, you may develop those connectors by following the guides in [Create your connector](../../recipes/create-your-connector/).

## Compositions

> _Connectors_ are stored both locally and remotely:

Reviewer 2: Seems that the connector data is divided into two parts, one part is hardcoded in the code, and another part is stored in the DB.

> As there are lots of properties in connector data, and to make the data loading and updating more efficient, we store part of connector data which will be modified frequently to DB and the rest of that locally.

Reviewer 2: Hard to read. I think so as well.

<img width="1324" alt="image" src="https://user-images.githubusercontent.com/10594507/176881244-bb033b08-f3d7-47e1-9fab-b889275509c5.png">

just suggestion:

```suggestion
There are lots of properties in connector data.
To make the connector data loading and updating more efficient, 
we store the frequently updated data in DB and the rest in the locality.
```

Author: Updated, good call.
